go-controller/pkg/kubevirt/dhcp.go

```diff
@@ -22,17 +22,62 @@ const (
 	dhcpLeaseTime = 3500
 )
 
+type DHCPConfigsOpt = func(*dhcpConfigs)
+
 type dhcpConfigs struct {
 	V4 *nbdb.DHCPOptions
 	V6 *nbdb.DHCPOptions
 }
 
+func WithIPv4Router(router string) func(*dhcpConfigs) {
+	return func(configs *dhcpConfigs) {
+		if configs.V4 == nil {
+			return
+		}
+		configs.V4.Options["router"] = router
+	}
+}
+
+func WithIPv4DNServer(dnsServer string) func(*dhcpConfigs) {
+	return func(configs *dhcpConfigs) {
+		if configs.V4 == nil {
+			return
+		}
+		configs.V4.Options["dns_server"] = dnsServer
+	}
+}
+
+func WithIPv6DNServer(dnsServer string) func(*dhcpConfigs) {
+	return func(configs *dhcpConfigs) {
+		if dnsServer == "" {
+			return
+		}
+		if configs.V6 == nil {
+			return
+		}
+		configs.V6.Options["dns_server"] = dnsServer
+	}
+}
+
 func EnsureDHCPOptionsForMigratablePod(controllerName string, nbClient libovsdbclient.Client, watchFactory *factory.WatchFactory, pod *corev1.Pod, ips []*net.IPNet, lsp *nbdb.LogicalSwitchPort) error {
+	dnsServerIPv4, dnsServerIPv6, err := retrieveDNSServiceClusterIPs(watchFactory)
+	if err != nil {
+		return fmt.Errorf("failed retrieving dns service cluster ip: %v", err)
+	}
+
+	return EnsureDHCPOptionsForLSP(controllerName, nbClient, pod, ips, lsp,
+		WithIPv4Router(ARPProxyIPv4),
```

Comment:
this is only for hypershift right ? For primary UDN, we would put here the GW, wouldn't we ?

Reply:
> this is only for hypershift right ? For primary UDN, we would put here the GW, wouldn't we ?

Yep, this is just a wrapper for `EnsureDHCPOptionsForMigratablePod` that is only called for the hypershift case.

For layer2 primary UDN is called directly at the secondaries controller.